rust/ql/lib/codeql/rust/elements/internal/ImplTraitTypeReprImpl.qll

```diff
@@ -21,5 +22,11 @@ module Impl {
    * //          ^^^^^^^^^^^^^^^^^^^^^^^^^^
    * ```
    */
-  class ImplTraitTypeRepr extends Generated::ImplTraitTypeRepr { }
+  class ImplTraitTypeRepr extends Generated::ImplTraitTypeRepr {
+    /**
+     * Gets the function for which this impl trait type occurs in the return
+     * type, if any.
+     */
+    Function getFunctionReturnPos() { this.getParentNode*() = result.getRetType().getTypeRepr() }
```

Comment:
I'd like us to come up with a better name for this predicate - in particular the pattern `get...Pos` makes my brain expect this to return a position / index, which it doesn't.  On the other hand "return position" is established phrasing so we don't want to lose that.  Perhaps just re-order it to `getReturnPosFunction`???

Reply:
What about `getReturnPosOfFunction`? We could also split this into two predicates `getFunction` and `isReturnPos`. Both of those seem clear, but we don't otherwise have a need for the general `getFunction` that also works for `impl` in arguments.